torch_geometric/explain/metrics.py

```diff
@@ -0,0 +1,125 @@
+from abc import ABC, abstractmethod
+from typing import Optional, Union
+
+import torch
+from torch import Tensor
+
+from torch_geometric.explain import Explainer, Explanation
+
+
+class ExplanationMetric(ABC):
+    r"""Abstract base class for explanation metrics."""
+    def __init__(self, explainer, index) -> None:
+        self.explainer = explainer
+        self.index = index
+
+    @abstractmethod
+    def __call__(self, explainer: Explainer, **kwargs):
+        r"""Computes the explanation metric for given explainer and explanation
+            Args:
+        explainer :obj:`~torch_geometric.explain.Explainer`
+            The explainer to evaluate
+        """
+
+    def get_inputs(self):
+        r"""Obtain inputs all different inputs over which to compute the
+        metrics."""
+
+    @abstractmethod
+    def compute_metric(self):
+        r"""Compute the metric over all inputs."""
+
+    @abstractmethod
+    def aggregate(self):
+        r"""Aggregate metrics over all inputs"""
+
+
+class Fidelity(ExplanationMetric):
+    r"""Fidelity+/- Explanation Metric as
+    defined in https://arxiv.org/abs/2206.09677"""
+    def __init__(self) -> None:
+        super().__init__()
+
+
+def fidelity(explainer: Explainer, explanation: Explanation,
+             target: Optional[Tensor] = None,
+             index: Optional[Union[int, Tensor]] = None,
+             output_type: str = 'raw', **kwargs):
+    r"""Evaluate the fidelity of Explainer and given
+    explanation produced by explainer
+
+    Args:
+        explainer :obj:`~torch_geometric.explain.Explainer`
+            The explainer to evaluate
+        explanation :obj:`~torch_teometric.explain.Explanation`
+            The explanation to evaluate
+        target (Tensor, optional): The target prediction, if not provided it
+            is inferred from obj:`explainer`, defaults to obj:`None`
+        index (Union[int, Tensor]): The explanation target index, for node-
+            and edge- level task it signifies the nodes/edges explained
+            respectively, for graph-level tasks it is assumed to be None,
+            defaults to obj:`None`
```

Comment:
```suggestion
        explanation :obj:`~torch_teometric.explain.Explanation`
            The explanation to evaluate
```
Do we need anything other than `explanation` now given other values are added in to `Explanation` in `Explainer`.